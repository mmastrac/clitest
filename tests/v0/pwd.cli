#!/usr/bin/env clitest --v0

using tempdir;

defer {
    $ ls -1 $PWD
    ! file

    cd $TEMP_DIR;

    $ rm -rf $PWD/subdir

    $ pwd
    *
}

$ pwd
%SET TEMP_DIR
! %{PATH}

$ mkdir subdir

# Avoid using `echo` because msys on Windows will munge the path
set PWD $PWD/subdir;

$ touch file
