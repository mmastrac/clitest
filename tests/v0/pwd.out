Running tests/v0/pwd.cli ...

using tempdir: <tmp>

pwd
---
<tmp>
---
✅ OK

mkdir subdir
---
(no output)
---
✅ OK

set PWD <tmp>/subdir

touch file
---
(no output)
---
✅ OK

(deferred) ls -1 $PWD
---
file
---
✅ OK

(deferred) cd <tmp>

(deferred) rm -rf $PWD/subdir
---
(no output)
---
✅ OK

(deferred) pwd
---
<tmp>
---
✅ OK

(cleanup) removing <tmp> && cd <root>/tests/v0

tests/v0/pwd.cli PASSED
1 test(s) passed
